frontend/src/components/FileUploader.js

```diff
@@ -8,68 +8,126 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 import { useDropzone } from 'react-dropzone';
-
 import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
-import { faTimes, faCircle } from '@fortawesome/free-solid-svg-icons';
-import { Tag, Button, Grid } from '@trussworks/react-uswds';
+import { faTrash } from '@fortawesome/free-solid-svg-icons';
+import { Button } from '@trussworks/react-uswds';
+import uploadFile from '../fetchers/File';
 
 import './FileUploader.css';
 
 function Dropzone(props) {
-  const { onChange } = props;
+  const { onChange, id, reportId } = props;
   const onDrop = (e) => {
+    let attachmentType;
+    if (id === 'attachments') {
+      attachmentType = 'ATTACHMENT';
+    } else if (id === 'other-resources') {
+      attachmentType = 'RESOURCE';
+    }
+    e.forEach(async (file) => {
+      try {
+        const data = new FormData();
+        data.append('reportId', reportId);
+        data.append('attachmentType', attachmentType);
+        data.append('file', file);
+        await uploadFile(data);
+      } catch (error) {
+        // eslint-disable-next-line no-console
```

Comment:
We'll want some kind of error to show to the user if we fail to upload the file. I can see a few ways to do it:

1) Move the `onDrop` definition into the `Dropzone` component. Update an error state in the `Dropzone` component if the `onDrop` method encounters an error. Display an error in the `Dropzone` component (probably an "Alert" above or below the `upload` button, [the side nav](https://github.com/adhocteam/Head-Start-TTADP/blob/main/frontend/src/components/Navigator/components/SideNav.js#L68) shows the general idea)
2) We define the `onDrop` callback in `FileUploader`, setting an error if needed, and passing that function down to the Dropzone component.
3) We define an `onError` in the `FileUploader` component and pass that as a prop alongside the `onChange` method. The `onDrop` method would then call `onError` if it encounters an error.

The first option seems like the best option to me because all the logic and callbacks are contained in one place, close to where the potential cause of the error is triggered.

Reply:
Done